src/.editorconfig

```diff
@@ -0,0 +1,138 @@
+# Rules in this file were initially inferred by Visual Studio IntelliCode from the C:\Repos\Acuminator\src codebase based on best match to current usage at 1/16/2022
+# You can modify the rules from these initially generated values to suit your own policies
+# You can learn more about editorconfig here: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
+[*.cs]
+
+
+#Core editorconfig formatting - indentation
+
+#use hard tabs for indentation
+indent_style = tab
```

Comment:
No war) I just picked idea to add .editorconfig file from @shawnburt 's PR but added one generated by VS because there were more settings specified. I suggest that it either deduced this from the code base or picked preferences from my VS settings. And my VS is configured this way because we use tabs in our main repo. 

I don't mind any style actually but I think keeping tabs here may have some advantages:

1. Tabs are already used in many places in Acuminator. Switching to space will result in inconsistencies in the indent style. 
2. Changing test source files for code fixes may bring unexpected test failures due to the changes in indentation done automatically by IDE.
3. No need to adapt to different indent style on switches between main repo and Acuminator

I don't mind switching to spaces, just don't want to update indentation in the whole code base.

Reply:
Well, at least in this PR it seems that all changed files use spaces and your new changes use tabs so the formatting becomes broken in GitHub.

We can leave test source files "as is", they are actually not related to the `.editorconfig` rules.

Reply:
For some weird reason my VS 2022 switched to spaces(((
Seems like weird settings sync issue - I remember setting indent settings for all languages.
Still I would use tabs since most of the files were untouched by this PR.

Are you sure that test sources are not affected by the editor.config file? The rules apply to [*.cs] files and test sources have .cs extension.
![image](https://user-images.githubusercontent.com/7687400/151858763-598312f8-dee8-4e67-95f9-b02df96b6d9d.png)

Reply:
We can ignore them with something like this (didn't check but found this approach on SO):

```
# Ignore files with actual and expected sources in the tests
[/src/Acuminator/Acuminator.Tests/**/Sources/*.cs]
charset = unset
end_of_line = unset
insert_final_newline = unset
trim_trailing_whitespace = unset
indent_style = unset
indent_size = unset
```

Reply:
As for the VS settings, ReSharper also messes it up — I have tabs in VS settings but it still puts spaces in some weird places.

Reply:
Btw, I approved the PR so you can merge it any time, my comments are rather minor.

Reply:
I will fix the indent first. Not good to introduce inconsistency.

> As for the VS settings, ReSharper also messes it up — I have tabs in VS settings but it still puts spaces in some weird places.
Actually I think VS deduced indentation setting for .editorconfig file from the code base. It left a comment about the settings being deduced from existing style. That's also the main reason why I'm for tabs - don't want to change existing style and introduce inconsistency. At least this should be done in another dedicated PR.